### PR TITLE
ev/walletconnect integration

### PR DESCRIPTION
**Pull request by vilsbole**
Replaces #2911 

### Steps 
1. Add a WallectConnect (WC) account to your dashboard.
2. Send a transaction from your account.

You should be able to go through the flow smoothly, and also have correct error handling when you reject a connection, a transaction, or modify the selected account or network within the mobile app.

---
Suggested WC mobile apps :
- Argent
- Metamask
- Trust

### CODE REVIEW

**Comment by LuckyLuker** [comment]
[ch4436]

**Comment by vilsbole** [comment]
> You're going to want to swap this out to also include WalletConnect because as the flow is now, you'll be sending the transaction to the walletconnect device, where it is broadcast, **then** requesting the user confirm the transaction, which just doesn't make much sense. It should be like this for all web3 wallets too, not just Metamask: https://github.com/MyCryptoHQ/MyCrypto/blob/master/common/v2/features/SendAssets/SendAssets.tsx#L86-L87
> 
> So maybe:
> 
> ```
> const alternativeStepsWallets = [ WalletId.METAMASK, WalletId.TRUST, WalletId.FRAME, WalletId.COINBASE, WalletId.WALLETCONNECT ];
> const walletId = senderAccount ? senderAccount.wallet : undefined;
> const steps = alternativeStepsWallets.includes(walletId) ? web3Steps : defaultSteps;
> ```

I updated the Send Flow to use the walletType. Now the path is determined using `isWeb3Wallet`

**Comment by vilsbole** [comment]
> It seems that txReceipt isn't set and thus, displayTxReceipt is undefined.
Solved. The useMemo in SendAssets was passing the old state to the step. Removed.

File: common/v2/components/SignTransactionWallets/WalletConnect.tsx [hunk 1]
```diff
@@ -0,0 +1,222 @@
...
+                on network {getNetworkByChainId(state.detectedChainId!, networks)!.name}
+              </Typography>
+            </SSection>
+            <SSection center={true} style={{ maxWidth: '430px' }}>
+              <Typography>Requesting signature for transcation:</Typography>
+              <CodeBlock>{objToString(rawTransaction)}</CodeBlock>
+            </SSection>
+          </>
+        ) : (
+          <SSection center={true}>
```
Comment (FrederikBolding) [inline common/v2/components/SignTransactionWallets/WalletConnect.tsx]: Use translation strings

File: common/v2/components/SignTransactionWallets/WalletConnect.tsx [hunk 2]
```diff
@@ -0,0 +1,222 @@
...
+                  inline={true}
+                  isCopyable={false}
+                  address={state.detectedAddress!}
+                  truncate={truncate}
+                />{' '}
+                on network {getNetworkByChainId(state.detectedChainId!, networks)!.name}
+              </Typography>
+            </SSection>
+            <SSection center={true} style={{ maxWidth: '430px' }}>
+              <Typography>Requesting signature for transcation:</Typography>
```
Comment (FrederikBolding) [inline common/v2/components/SignTransactionWallets/WalletConnect.tsx]: Use translation strings

File: common/v2/components/SignTransactionWallets/WalletConnect.tsx [hunk 3]
```diff
@@ -0,0 +1,222 @@
...
+    <>
+      <SHeader>
+        {translate('SIGNER_SELECT_WALLETCONNECT', { $walletId: translateRaw('X_WALLETCONNECT') })}
+      </SHeader>
+      <SContent>
+        {state.isConnected ? (
+          <>
+            <SSection center={true}>
+              <Typography>
+                Session connected with{' '}
```
Comment (FrederikBolding) [inline common/v2/components/SignTransactionWallets/WalletConnect.tsx]: Use translation strings

File: common/v2/components/SignTransactionWallets/WalletConnect.tsx [hunk 4]
```diff
@@ -0,0 +1,222 @@
...
+            absolute={true}
+            center={true}
+            show={state.isConnected || !R.isEmpty(state.errors)}
+          >
+            <SContainer>
+              {!R.isEmpty(state.errors) ? (
+                getErrorMessage(state.errors![0])
+              ) : (
+                <>
+                  <Typography>Pending Confirmation</Typography>
```
Comment (FrederikBolding) [inline common/v2/components/SignTransactionWallets/WalletConnect.tsx]: Use translation strings

File: common/v2/components/SignTransactionWallets/WalletConnect.tsx [hunk 5]
```diff
@@ -0,0 +1,222 @@
...
+                  <div style={{ margin: '1em 0' }}>
+                    <Spinner />
+                  </div>
+                  <Typography>Please confirm the transaction on your wallet</Typography>
```
Comment (FrederikBolding) [inline common/v2/components/SignTransactionWallets/WalletConnect.tsx]: Use translation strings

File: common/v2/components/WalletUnlock/WalletConnect.tsx [hunk 6]
```diff
@@ -0,0 +1,118 @@
...
+
+  useEffect(() => {
+    if (!state.detectedAddress) return;
+    onUnlock(WalletService.init(state.detectedAddress));
+  }, [state.detectedAddress]);
+
+  return (
+    <>
+      <SHeader>
+        {translate('UNLOCK_WALLET')} {`Your ${translateRaw('X_WALLETCONNECT')} device`}
```
Comment (FrederikBolding) [inline common/v2/components/WalletUnlock/WalletConnect.tsx]: Use translation strings

File: common/v2/components/WalletUnlock/WalletConnect.tsx [hunk 7]
```diff
@@ -0,0 +1,118 @@
...
+        <SSection center={true} withOverlay={true}>
+          <Overlay
+            absolute={true}
+            center={true}
+            show={state.isConnected || !R.isEmpty(state.errors)}
+          >
+            <SContainer>
+              {!R.isEmpty(state.errors) && (
+                <>
+                  <Typography>Session Rejected</Typography>
```
Comment (FrederikBolding) [inline common/v2/components/WalletUnlock/WalletConnect.tsx]: Use translation strings

File: common/v2/components/WalletUnlock/WalletConnect.tsx [hunk 8]
```diff
@@ -0,0 +1,118 @@
...
+                  <div style={{ marginTop: '1em' }}>
+                    <Button onClick={requestConnection}>Try Again</Button>
```
Comment (FrederikBolding) [inline common/v2/components/WalletUnlock/WalletConnect.tsx]: Use translation strings

File: common/v2/features/Toasts/constants.ts [hunk 9]
```diff
@@ -45,5 +46,14 @@ export const toastConfigs: ToastConfigsProps = {
     message: () => 'All systems good to go.',
     type: ToastType.SUCCESS,
     position: 'top-left'
+  },
+  [ToastTemplates.walletConnectReject]: {
+    header: 'Connection request rejected',
```
Comment (FrederikBolding) [inline common/v2/features/Toasts/constants.ts]: Use translation strings

Comment (vilsbole) [reply common/v2/features/Toasts/constants.ts]: the toast is out of service so I removed it.

File: common/v2/components/WalletUnlock/WalletConnect.tsx (hunk 7 above)
Comment (vilsbole) [reply common/v2/components/WalletUnlock/WalletConnect.tsx]: Done

File: common/v2/components/WalletUnlock/WalletConnect.tsx (hunk 6 above)
Comment (vilsbole) [reply common/v2/components/WalletUnlock/WalletConnect.tsx]: Done

File: common/v2/components/SignTransactionWallets/WalletConnect.tsx (hunk 5 above)
Comment (vilsbole) [reply common/v2/components/SignTransactionWallets/WalletConnect.tsx]: Done

File: common/v2/components/SignTransactionWallets/WalletConnect.tsx (hunk 4 above)
Comment (vilsbole) [reply common/v2/components/SignTransactionWallets/WalletConnect.tsx]: Done

File: common/v2/components/SignTransactionWallets/WalletConnect.tsx (hunk 3 above)
Comment (vilsbole) [reply common/v2/components/SignTransactionWallets/WalletConnect.tsx]: Done

File: common/v2/components/SignTransactionWallets/WalletConnect.tsx (hunk 2 above)
Comment (vilsbole) [reply common/v2/components/SignTransactionWallets/WalletConnect.tsx]: Done

File: common/v2/components/SignTransactionWallets/WalletConnect.tsx (hunk 1 above)
Comment (vilsbole) [reply common/v2/components/SignTransactionWallets/WalletConnect.tsx]: Done

**Pull request merged**